src/service/http-server.integration.test.ts

```diff
@@ -0,0 +1,232 @@
+    const url = `${getGrafanaEndpoint(domain)}/${dashboardUid}?panelId=${panelIds.table}&render=1&from=1699333200000&to=1699344000000`;
+    const response = await request(server.app)
+      .get(
+        `/render?url=${encodeURIComponent(
+          url
+        )}&timeout=5&renderKey=${renderKey}&domain=${domain}&width=${imageWidth}&height=${imageHeight}&deviceScaleFactor=1`
+      )
+      .set('X-Auth-Token', '-');
+
+    expect(response.statusCode).toEqual(200);
+    expect(response.headers['content-type']).toEqual('image/png');
+
+    const pixelDiff = compareImage('table', response.body);
+    expect(pixelDiff).toBeLessThan(imageDiffThreshold);
+  });
+
+  it('should respond with a panel error screenshot', async () => {
+    const url = `${getGrafanaEndpoint(domain)}/${dashboardUid}?panelId=${panelIds.error}&render=1&from=1699333200000&to=1699344000000`;
+    const response = await request(server.app)
+      .get(
+        `/render?url=${encodeURIComponent(
+          url
+        )}&timeout=5&renderKey=${renderKey}&domain=${domain}&width=${imageWidth}&height=${imageHeight}&deviceScaleFactor=1`
+      )
+      .set('X-Auth-Token', '-');
+
+    expect(response.statusCode).toEqual(200);
+    expect(response.headers['content-type']).toEqual('image/png');
+
+    const pixelDiff = compareImage('error', response.body);
+    expect(pixelDiff).toBeLessThan(imageDiffThreshold);
+  });
+
+  // TODO: this doesn't work on Drone
+  // it('should timeout when a panel is too slow to load', async () => {
+  //   const url = `${getGrafanaEndpoint(domain)}/${dashboardUid}?panelId=${panelIds.slow}&render=1&from=1699333200000&to=1699344000000`;
+  //   const response = await request(server.app)
+  //     .get(
+  //       `/render?url=${encodeURIComponent(
+  //         url
+  //       )}&timeout=5&renderKey=${renderKey}&domain=localhost&width=${imageWidth}&height=${imageHeight}&deviceScaleFactor=1`
+  //     )
+  //     .set('X-Auth-Token', '-');
+
+  //   expect(response.statusCode).toEqual(200);
+  //   expect(response.headers['content-type']).toEqual('image/png');
+
+  //   const pixelDiff = compareImage('slow', response.body);
+  //   expect(pixelDiff).toBeLessThan(imageDiffThreshold);
+  // }, 30000);
+});
+
+// compareImage returns the number of different pixels between the image stored in the test file and the one from the response body.
+// It updates the stored file and returns 0 if tests are run with UPDATE_GOLDEN=true.
+// It writes the diff file to /testdata if tests are run with SAVE_DIFF=true.
+function compareImage(testName: string, responseBody: any): number {
+  const goldenFilePath = path.join(goldenFilesFolder, `${testName}.png`);
+  if (process.env['UPDATE_GOLDEN'] === 'true') {
+    fs.writeFileSync(goldenFilePath, responseBody);
+    return 0;
+  }
+
+  let diff: { width: number; height: number; data: Uint8ClampedArray } | null = null;
+  if (process.env['SAVE_DIFF'] === 'true') {
```

Comment:
I saw we are setting some env files at the beginning inside `beforeAll` maybe we can read the env value there only once and create a variable to reference it here, similar to how we are setting up `domain`.

Reply:
I updated the code to do that 👍